app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,47 @@
+class LikesController < ApplicationController
+
+  before_action :authenticate_user!, :except => [:index, :show]
+
+  def new
+    @post = Post.find(params[:post_id])
+    @like = Like.new
+  end
+
+  def create
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for create is too high. [20.2/15] (http://c2.com/cgi/wiki?AbcMetric)